contracts/evoting/evoting.go

```diff
@@ -471,6 +474,143 @@ func (e evotingCommand) closeElection(snap store.Snapshot, step execution.Step)
 	return nil
 }
 
+// registerPubShares implements commands.It performs the REGISTER_PUB_SHARES command
+func (e evotingCommand) registerPubShares(snap store.Snapshot, step execution.Step) error {
+	var tx types.RegisterPubSharesTransaction
+
+	err := getTransaction(step.Current, &tx)
+	if err != nil {
+		return xerrors.Errorf(errGetTransaction, err)
+	}
+
+	err = checkPreviousPubSharesTransactions(step, tx.Round)
```

Comment:
Ok ! I added it because I had some troubles with adding the pubShares, all the node send their tx, all tx are accepted but in the end we only have on set of pubShares...